Sources/HummingbirdCore/Response/ResponseWriter.swift

```diff
@@ -0,0 +1,87 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Hummingbird server framework project
+//
+// Copyright (c) 2024 the Hummingbird authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See hummingbird/CONTRIBUTORS.txt for the list of Hummingbird authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import HTTPTypes
+import NIOCore
+import NIOHTTPTypes
+
+/// ResponseWriter that writes directly to AsyncChannel
+public struct ResponseWriter: ~Copyable {
+    @usableFromInline
+    let outbound: NIOAsyncChannelOutboundWriter<HTTPResponsePart>
+
+    /// Write HTTP head part and return ``ResponseBodyWriter`` to write response body
+    ///
+    /// - Parameter head: Response head
+    /// - Returns: Response body writer used to write HTTP response body
+    @inlinable
+    public consuming func writeHead(_ head: HTTPResponse) async throws -> some ResponseBodyWriter {
+        try await self.outbound.write(.head(head))
+        return RootResponseBodyWriter(outbound: self.outbound)
+    }
+
+    /// Write Informational HTTP head part
+    ///
+    /// Calling this with a non informational HTTP response head will cause a precondition error
+    /// - Parameter head: Informational response head
+    @inlinable
+    public func writeInformationalHead(_ head: HTTPResponse) async throws {
```

Comment:
Is this not consuming?

Reply:
No, you need to be able to send the actual header after the informational header.